From be73c196eb555d93be9afc1db3277a99e7a3d1c9 Mon Sep 17 00:00:00 2001
From: Mika Laitio <mika.laitio@amd.com>
Date: Thu, 9 Oct 2025 09:29:30 -0700
Subject: [PATCH 1/2] rockbuilder vllm dependency fixes

- remove torch version and download urls
  because that caused the therock version of
  rocm sdk to be replaced with the one
  version in rocm 6.2.4 repository
- remove triton version requirements to
  3.2 to allow building a never triton version
  (pytorch 2.7.0 will by default build
   triton 3.3.0 in ci scripts)
- remove amdsmi version requirement
- update pyprojects.toml for install time requirements

Signed-off-by: Mika Laitio <mika.laitio@amd.com>
---
 pyproject.toml              |  4 ++--
 requirements/common.txt     |  2 +-
 requirements/rocm-build.txt | 11 +++++------
 3 files changed, 8 insertions(+), 9 deletions(-)

diff --git a/pyproject.toml b/pyproject.toml
index 416423abc..f7ff63461 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -4,9 +4,9 @@ requires = [
     "cmake>=3.26.1",
     "ninja",
     "packaging>=24.2",
-    "setuptools>=77.0.3,<80.0.0",
+    "setuptools",
     "setuptools-scm>=8.0",
-    "torch == 2.8.0",
+    "torch",
     "wheel",
     "jinja2",
 ]
diff --git a/requirements/common.txt b/requirements/common.txt
index b8665104b..e9d0385a3 100644
--- a/requirements/common.txt
+++ b/requirements/common.txt
@@ -36,7 +36,7 @@ mistral_common[image,audio] >= 1.8.2
 opencv-python-headless >= 4.11.0    # required for video IO
 pyyaml
 six>=1.16.0; python_version > '3.11' # transitive dependency of pandas that needs to be the latest version for python 3.12
-setuptools>=77.0.3,<80; python_version > '3.11' # Setuptools is used by triton, we need to ensure a modern version is installed for 3.12+ so that it does not try to import distutils, which was removed in 3.12
+setuptools; python_version > '3.11' # Setuptools is used by triton, we need to ensure a modern version is installed for 3.12+ so that it does not try to import distutils, which was removed in 3.12
 einops # Required for Qwen2-VL.
 compressed-tensors == 0.11.0 # required for compressed-tensors
 depyf==0.19.0 # required for profiling and debugging with compilation config
diff --git a/requirements/rocm-build.txt b/requirements/rocm-build.txt
index affe562c2..5eb2cb7bb 100644
--- a/requirements/rocm-build.txt
+++ b/requirements/rocm-build.txt
@@ -2,15 +2,14 @@
 -r common.txt
 
 --extra-index-url https://download.pytorch.org/whl/rocm6.3
-torch==2.8.0
-torchvision==0.23.0
-torchaudio==2.8.0
+torch
+torchvision
+torchaudio
 
-triton==3.3.0
+triton
 cmake>=3.26.1,<4
 packaging>=24.2
-setuptools>=77.0.3,<80.0.0
+setuptools
 setuptools-scm>=8
 wheel
 jinja2>=3.1.6
-amdsmi==6.2.4
-- 
2.43.0

